url748e34f967_Diab1o_java_7z_archiver_tgz-pJ8-com_swemel_sevenzip_TestJ8/src/com/swemel/sevenzip/archive/SevenZipFolderInStream.java

```diff
@@ -0,0 +1,107 @@
+package com.swemel.sevenzip.archive;
+
+import com.swemel.sevenzip.UpdateItem;
+import com.swemel.sevenzip.common.InStreamWithCRC;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SevenZipFolderInStream extends InputStream {
+    private InStreamWithCRC stream;
+
+    private boolean _fileIsOpen;
+    private long _filePos;
+    private List<UpdateItem> updateItems;
+    private int _numFiles;
+    private int _fileIndex;
+    private int off;
+
+    private final List<Integer> CRCs = new ArrayList<>();
+    private final List<Long> sizes = new ArrayList<>();
+
+
+    public void init(List<UpdateItem> updateItems, int off, int numFiles) {
+        this.updateItems = updateItems;
+        _numFiles = numFiles;
+        _fileIndex = 0;
+        this.off = off;
+        _fileIsOpen = false;
+    }
+
+    private void openStream() throws FileNotFoundException {
+        _filePos = 0;
+        while (_fileIndex < _numFiles) {
+            stream = new InStreamWithCRC(updateItems.get(off + _fileIndex).getFullName());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *stream*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).